### ISSUE TITLE
GraphQL 1.9.4 incompatible with Ruby 2.2

### ISSUE THREAD

**Issue by delner** [rmosolgo/graphql-ruby#2235]
I have some tests that use `graphql` 1.9.4, and produce the following error with Ruby 2.2.10:

```
Failure/Error: require 'graphql'

SyntaxError:
  /usr/local/bundle/gems/graphql-1.9.4/lib/graphql/static_validation/base_visitor.rb:149: syntax error, unexpected '.'
            is_list = @argument_definitions.last&.type&.list? == true
                                                  ^
# /usr/local/bundle/gems/graphql-1.9.4/lib/graphql/static_validation.rb:8:in `require'
# /usr/local/bundle/gems/graphql-1.9.4/lib/graphql/static_validation.rb:8:in `<top (required)>'
# /usr/local/bundle/gems/graphql-1.9.4/lib/graphql.rb:96:in `require'
```

Looks like there's use of the safe operator which isn't defined until Ruby 2.3 http://mitrev.net/ruby/2015/11/13/the-operator-in-ruby/

I think the gemspec should be updated to reflect Ruby 2.3+ support only?

**Comment by rmosolgo** [comment]
Oh my, I think I bumped travis.yml when I was fighting with bundler versions ... Oh, yes, here it is: 

```
~/code/graphql-ruby $ rbenv local 2.2.8
~/code/graphql-ruby $ bundle install
You must use Bundler 2 or greater with this lockfile.
~/code/graphql-ruby $ gem install bundler
Fetching: bundler-2.0.1.gem (100%)
ERROR:  Error installing bundler:
	bundler requires Ruby version >= 2.3.0.
```

So, Bundler 2+ only supports Ruby 2.3+. Well, it looks like 2.2 _was_ EOL'd, so I'm ok with dropping support for it. 

Do you need Ruby 2.2 support? I don't mind support it if we can get it passing on travis (ie reverting 2850bab1aa6f9a682a0e54245c786230b0530d71 and then doing ... whatever ... is necessary to make the build pass again). I had a hard time getting the right Bundler version on there, though.

**Comment by delner** [comment]
Strictly speaking, no, I don't use Ruby 2.2, but I do maintain a library which does and implements features around `graphql`. It might be reasonable to drop 2.2 support if its now EOL, as long as the gemspec reflects this requirement for 2.3+.

**Comment by rmosolgo** [comment]
Fixed by #2302

**Comment by delner** [comment]
Thanks for the update @rmosolgo!
